proposals/0000-swiftpm-script-support.md

```diff
@@ -0,0 +1,141 @@
+
+## Proposed solution
+
+We propose to add a new attribute @package to the import statement for referencing Swift packages. This attribute would be only valid when running in the Swift interpreter (via `swift MyScript.swift` or `swift run MyScript.swift`) and will be rejected if used during compilation of a regular Swift module. Swift Package Manager will then fetch and build the declared packages and make them available to the script when executing it.
+
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+This script can be executable by either `swift MyScript.swift` or `swift run MyScript.swift` which will utilize SwiftPM's dependency fetching and resolution logic.
+
+The `@package(...)`syntax will allow the same parameters as SwiftPM's [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel natural to developers familiar with Swift Package Manager manifest syntax.  
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above Package descriptions, the `SwiftToolsSupport` product does not match a target within the package description it must be specified explicitly.
+
+```swift
+@package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.0.1"), products: ["SwiftToolsSupport"])
+import TSCBasic 
+
+```
+
+#### Dependency version requirements
+
+The `@package(..)` annotation would also allow a developer to specify requirements to ensure your scripts are reproducible. The syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+*Revision*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .revision("5fa313eae1ca127ad3c706e14c564399989cb1b1")) 
+import Yams 
+```
+
+*Branch*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .branch("releases/1.0")) 
+import Yams 
+```
+
+*Exact Version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .exact("2.0.0")) 
+import Yams 
+```
+
+*Range of version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .from("1.0")) 
```

Comment:
This should be: `@package(url: "https://github.com/jpsim/Yams.git", .from: "1.0.0")`